app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -117,8 +117,13 @@ class BusinessEntityController @Inject() (
       journeyTraverseService.getPrevious[BusinessEntityJourney].flatMap { previous =>
         store.get[BusinessEntityJourney].flatMap {
           case Some(journey) =>
-            val form = journey.eori.value.fold(eoriForm)(eori => eoriForm.fill(FormValues(eori)))
-            Ok(eoriPage(form, previous)).toFuture
+            val isEligible = journey.addBusiness.value.nonEmpty
```

Comment:
can we move this val in the journey?

Reply:
Yeah my thoughts exactly. I've gone some way in this direction with the undertaking journey. Ideally we can just ask the journey what's next. Cleans up the controllers and makes it easier to unit test the journey logic.